components/script/dom/htmltableelement.rs

```diff
@@ -398,16 +398,20 @@ impl HTMLTableElementMethods for HTMLTableElement {
     // https://html.spec.whatwg.org/multipage/#dom-table-deleterow
     fn DeleteRow(&self, mut index: i32) -> Fallible<()> {
         let rows = self.Rows();
+
         // Step 1.
-        if index == -1 {
-            index = rows.Length() as i32 - 1;
-        }
-        // Step 2.
-        if index < 0 || index as u32 >= rows.Length() {
+        if index < -1 || index >= rows.Length() as i32 {
             return Err(Error::IndexSize);
         }
-        // Step 3.
-        DomRoot::upcast::<Node>(rows.Item(index as u32).unwrap()).remove_self();
+
+        // Step 2
+        if index == -1 && rows.Length() > 0 {
+            index = rows.Length() as i32 - 1;
+
+            // Step 3.
+            DomRoot::upcast::<Node>(rows.Item(index as u32).unwrap()).remove_self();
+        }
+
```

Comment:
I think this makes it so that if the index passed isn't -1 no row is removed. It might be simpler to make this more like:

```suggestion
            if num_rows == 0 {
                return Ok(());
             }
             
             if index == -1 {
                 index = num_rows - 1;
             }

             // Step 3: Otherwise, remove the indexth element in the rows collection from its parent.
             DomRoot::upcast::<Node>(rows.Item(index as u32).unwrap()).remove_self()
        }
```